tempodb/search/backend_search_block.go

```diff
@@ -23,6 +26,70 @@ type BackendSearchBlock struct {
 	l        *local.Backend
 }
 
+//nolint:golint
+type SearchDataCombiner struct{}
+
+func (*SearchDataCombiner) Combine(_ string, searchData ...[]byte) ([]byte, bool) {
+
+	if len(searchData) <= 0 {
+		return nil, false
+	}
+
+	if len(searchData) == 1 {
+		return searchData[0], false
+	}
+
+	// Squash all datas into 1
+	data := tempofb.SearchEntryMutable{}
+	kv := &tempofb.KeyValues{} // buffer
+	for _, sb := range searchData {
+		sd := tempofb.SearchEntryFromBytes(sb)
+		for i := 0; i < sd.TagsLength(); i++ {
+			sd.Tags(kv, i)
+			for j := 0; j < kv.ValueLength(); j++ {
+				data.AddTag(string(kv.Key()), string(kv.Value(j)))
+			}
+		}
+		data.SetStartTimeUnixNano(sd.StartTimeUnixNano())
+		data.SetEndTimeUnixNano(sd.EndTimeUnixNano())
+		data.TraceID = sd.Id()
+	}
+
+	return data.ToBytes(), true
+}
+
+var _ common.ObjectCombiner = (*SearchDataCombiner)(nil)
+
+//nolint:golint
+type SearchDataIterator struct {
+	currentIndex int
+	records      []common.Record
+	file         *os.File
+}
+
+func (s *SearchDataIterator) Next(_ context.Context) (common.ID, []byte, error) {
+	if s.currentIndex >= len(s.records) {
+		return nil, nil, io.EOF
+	}
+
+	currentRecord := s.records[s.currentIndex]
+	buffer := make([]byte, currentRecord.Length)
```

Comment:
Should try to resuse the buffer here.  Not blocking, I can follow up in another PR.